### PR TITLE
[SPARK-47629][INFRA]  Add `common/variant` and `connector/kinesis-asl` to maven daily test module list

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr add `common/variant` and `connector/kinesis-asl` to maven daily test module list.

### Why are the changes needed?
Synchronize the modules to be tested in Maven daily test

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Monitor GA after merge

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -327,6 +327,11 @@ jobs:
             ${{ needs.precondition.outputs.image_url }}
           # Use the infra image cache to speed up
           cache-from: type=registry,ref=ghcr.io/apache/spark/apache-spark-github-action-image-cache:${{ inputs.branch }}
+  maven-test:
```
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:330]: will revert after test

**Review by HyukjinKwon — APPROVED** [review]
LGTM if tests pass

**Comment by LuciferYang** [comment]
```
2024-03-28T18:48:08.9171911Z 3 warnings.
2024-03-28T18:48:12.6489322Z Discovery starting.
2024-03-28T18:48:14.4086181Z Discovery completed in 1 second, 814 milliseconds.
2024-03-28T18:48:14.4182932Z Run starting. Expected test count is: 35
2024-03-28T18:48:14.4248382Z KinesisInputDStreamBuilderSuite:
2024-03-28T18:48:14.4712987Z - should raise an exception if the StreamingContext is missing
2024-03-28T18:48:14.4756297Z - should raise an exception if the stream name is missing
2024-03-28T18:48:14.4758553Z - should raise an exception if the checkpoint app name is missing
2024-03-28T18:48:14.6237137Z - should propagate required values to KinesisInputDStream
2024-03-28T18:48:14.6309982Z - should propagate default values to KinesisInputDStream
2024-03-28T18:48:14.8843092Z OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
2024-03-28T18:48:15.4918194Z - should propagate custom non-auth values to KinesisInputDStream
2024-03-28T18:48:15.4980613Z - old Api should throw UnsupportedOperationExceptionexception with AT_TIMESTAMP
2024-03-28T18:48:15.7380623Z WithoutAggregationKinesisBackedBlockRDDSuite:
2024-03-28T18:48:15.7397598Z - Basic reading from Kinesis [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:15.7401340Z - Read data available in both block manager and Kinesis [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:15.7404229Z - Read data available only in block manager, not in Kinesis [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:15.7407274Z - Read data available only in Kinesis, not in block manager [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:15.7411087Z - Read data available partially in block manager, rest in Kinesis [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:15.7413957Z - Test isBlockValid skips block fetching from block manager [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:15.7416964Z - Test whether RDD is valid after removing blocks from block manager [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:15.8027369Z KinesisReceiverSuite:
2024-03-28T18:48:16.0598578Z - process records including store and set checkpointer
2024-03-28T18:48:16.0699421Z - split into multiple processes if a limitation is set
2024-03-28T18:48:16.0819317Z - shouldn't store and update checkpointer when receiver is stopped
2024-03-28T18:48:16.0934673Z - shouldn't update checkpointer when exception occurs during store
2024-03-28T18:48:16.1025986Z - shutdown should checkpoint if the reason is TERMINATE
2024-03-28T18:48:16.1071125Z - shutdown should not checkpoint if the reason is something other than TERMINATE
2024-03-28T18:48:16.1126724Z - retry success on first attempt
2024-03-28T18:48:16.1197799Z - retry success on second attempt after a Kinesis throttling exception
2024-03-28T18:48:16.1800756Z - retry success on second attempt after a Kinesis dependency exception
2024-03-28T18:48:16.1858656Z - retry failed after a shutdown exception
2024-03-28T18:48:16.1921030Z - retry failed after an invalid state exception
2024-03-28T18:48:16.1961305Z - retry failed after unexpected exception
2024-03-28T18:48:16.3093821Z - retry failed after exhausting all retries
2024-03-28T18:48:16.4074362Z WithAggregationKinesisBackedBlockRDDSuite:
2024-03-28T18:48:16.4079349Z - Basic reading from Kinesis [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:16.4083548Z - Read data available in both block manager and Kinesis [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:16.4087178Z - Read data available only in block manager, not in Kinesis [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:16.4090667Z - Read data available only in Kinesis, not in block manager [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:16.4094266Z - Read data available partially in block manager, rest in Kinesis [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:16.4097683Z - Test isBlockValid skips block fetching from block manager [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:16.4101297Z - Test whether RDD is valid after removing blocks from block manager [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:16.4733873Z WithAggregationKinesisStreamSuite:
2024-03-28T18:48:16.5597475Z - RDD generation
2024-03-28T18:48:16.6061947Z - basic operation [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:16.6064494Z - custom message handling [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:16.7100481Z - Kinesis read with custom configurations
2024-03-28T18:48:16.7333239Z - split and merge shards in a stream [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:16.7334445Z - failure recovery [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:16.7335501Z - Prepare KinesisTestUtils [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:16.7821666Z KinesisCheckpointerSuite:
2024-03-28T18:48:16.7831058Z - checkpoint is not called twice for the same sequence number
2024-03-28T18:48:16.7838199Z - checkpoint is called after sequence number increases
2024-03-28T18:48:16.8023248Z - should checkpoint if we have exceeded the checkpoint interval
2024-03-28T18:48:16.8063902Z - shouldn't checkpoint if we have not exceeded the checkpoint interval
2024-03-28T18:48:16.8140028Z - should not checkpoint for the same sequence number
2024-03-28T18:48:16.8179612Z - removing checkpointer checkpoints one last time
2024-03-28T18:48:16.8907449Z - if checkpointing is going on, wait until finished before removing and checkpointing
2024-03-28T18:48:16.9456192Z SparkAWSCredentialsBuilderSuite:
2024-03-28T18:48:16.9457552Z - should build DefaultCredentials when given no params
2024-03-28T18:48:16.9458711Z - should build BasicCredentials
2024-03-28T18:48:16.9461090Z - should build STSCredentials
2024-03-28T18:48:16.9470237Z - SparkAWSCredentials classes should be serializable
2024-03-28T18:48:16.9978309Z WithoutAggregationKinesisStreamSuite:
2024-03-28T18:48:17.0384093Z - RDD generation
2024-03-28T18:48:17.0716143Z - basic operation [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:17.0718145Z - custom message handling [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:17.1532701Z - Kinesis read with custom configurations
2024-03-28T18:48:17.1870008Z - split and merge shards in a stream [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:17.1876655Z - failure recovery [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:17.1879241Z - Prepare KinesisTestUtils [enable by setting env var ENABLE_KINESIS_TESTS=1] !!! IGNORED !!!
2024-03-28T18:48:17.2167546Z Run completed in 4 seconds, 599 milliseconds.
2024-03-28T18:48:17.2168692Z Total number of tests run: 35
2024-03-28T18:48:17.2169621Z Suites: completed 9, aborted 0
2024-03-28T18:48:17.2170604Z Tests: succeeded 35, failed 0, canceled 0, ignored 24, pending 0
2024-03-28T18:48:17.2171403Z All tests passed.
```

https://github.com/LuciferYang/spark/actions/runs/8468321679/job/23201055867

kinesisI-asl already passed

**Comment by LuciferYang** [comment]
The `variant ` module does not have any test cases now. Let me revert the change of `build_and_test.yml`

**Comment by LuciferYang** [comment]
Maven test passed. Merged into master for Apache Spark 4.0. I will monitor GA ~ 

Thanks @HyukjinKwon

**Pull request closed without merging**